doc/manual.rst

```diff
@@ -3084,10 +3084,24 @@ exceptions:
   translated by the compiler, the other statements are not checked for
   semantics! However, each condition is checked for semantics.
 
-The `when` statement enables conditional compilation techniques. As
-a special syntactic extension, the `when` construct is also available
+The `when` statement enables conditional compilation techniques.
+
+When statement inside `object`
+------------------------------
+
+As a special syntactic extension, the `when` construct is also available
 within `object` definitions.
 
+At which point the `when` construct is evaluated and collapsed depends on
+whether the object has generic parameters:
+* if it does not, the conditions are evaluated and a branch is chosen
```

Comment:
Hmm, I wonder if this is a case of a null/zero-ary generic?

To clarify what a null/zero-ary generic is:
```nim
type
  Foo = object
  Foo[] = object
```

Although very similar, the second `Foo[]` is not really equivalent and a generic with zero your parameters. The latter triggers instantiation.

Reply:
Ah, darn, I forgot about nullary generics again :D. For consistency with generic procedures, where a nullary generic procedure requires instantiation, I think the same should be true for types.

I'll adjust the specification to be explicit about nullary generic types (the implementation already handles nullary generics like you described).

Reply:
I've updated the specification to account for nullary generic types.

Reply:
nullary generics really only exist to:
1. support generic paratial application and "type parameter subtraction" like scenarios
2. trick us

most of the time it's the latter. 😆 

---

imma read over the commit now.